runtime/mount/kernel.py

```diff
@@ -767,137 +772,202 @@ async def save_kernel_snapshot(self) -> None:
+
+            s_anndata = list(self.ann_data_objects.keys())
+
+            s_depens = {
+                "s_globals": s_globals,
+                "s_nodes": s_nodes,
+                "s_signals": s_signals,
+                "widget_signals": {k: v.id for k, v in self.widget_signals.items()},
+                "nodes_with_widgets": list(self.nodes_with_widgets.keys()),
+                "cell_rnodes": {k: v.id for k, v in self.cell_rnodes.items()},
+                "s_anndata": s_anndata,
+                # todo(kenny): figure out what to do with these
+                "ldata_dataframes": {},
+                "registry_dataframes": {},
+                "url_dataframes": {},
+            }
 
-        s_globals: dict[str, SerializedSignal | SerializedGlobal] = {}
-        for k, val in self.k_globals.items():
-            if k == "__builtins__":
-                continue
-            if isinstance(val._value, Signal):
-                s_globals[k] = val._value.serialize()
-            elif isinstance(val._value, BaseWidget):
-                assert val._value._signal.id in s_signals
-                s_globals[k] = val._value.serialize()
-            elif isinstance(val._value, ad.AnnData):
-                s_globals[k] = serialize_anndata(val._value, snapshot_dir)
-            else:
-                s_val, msg = safe_serialize_obj(val._value)
-                s_globals[k] = {"value": s_val, "error_msg": msg}
-
-        s_anndata = list(self.ann_data_objects.keys())
-
-        s_depens = {
-            "s_globals": s_globals,
-            "s_nodes": s_nodes,
-            "s_signals": s_signals,
-            "widget_signals": {k: v.id for k, v in self.widget_signals.items()},
-            "nodes_with_widgets": list(self.nodes_with_widgets.keys()),
-            "cell_rnodes": {k: v.id for k, v in self.cell_rnodes.items()},
-            "s_anndata": s_anndata,
-            # todo(kenny): figure out what to do with these
-            "ldata_dataframes": {},
-            "registry_dataframes": {},
-            "url_dataframes": {},
-        }
+            data = orjson.dumps(s_depens)
+        except Exception:
+            await self.send({"type": "save_kernel_snapshot",
+                             "status": "error",
+                             "data": {"error_msg": traceback.format_exc()}})
+            return
 
-        (snapshot_dir / snapshot_f_name).write_text(
-            orjson.dumps(s_depens).decode("utf-8")
-        )
+        total = len(data)
+        await self.send({"type": "save_kernel_snapshot",
+                         "status": "start",
+                         "data": {"total_bytes": total}})
+
+        with (snapshot_dir / snapshot_f_name).open("wb") as f:
+            sent_since_last = 0
+            for start in range(0, total, SNAPSHOT_CHUNK):
+                end = min(start + SNAPSHOT_CHUNK, total)
+                f.write(data[start:end])
+                os.fsync(f.fileno())
```

Comment:
We are gonna close the file anyways after the context manager exits.... why do we need to fsync here?

Reply:
leftover from something was trying agree it has no function